ui/src/services/uiSettings.ts

```diff
@@ -13,7 +13,7 @@ export class UiSettings {
   public static settings: Settings | null = null
 
   private static promise: Promise<Settings> | null = null
-  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : BASE_URL() ?? window.location.origin
+  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : ''
```

Comment:
Can you explain this change? Seems odd to have no base url when not in development.

Reply:
Yup -
`BASE_URL` is the base url of static assets built when we call `npm run build` and configured in `vite.config.ts`. The `/ui-settings` endpoint is served from the same base route as the UI app itself _minus any subdirectory we mount it to_. It's a somewhat special case because the settings endpoint is served from `/ui-settings` but the UI app is mounted to `/{PREFECT_UI_SERVE_BASE}`. When this is set to `BASE_URL`, the UI incorrectly attempts to fetch settings from the asset base url, which resolves to `/{PREFECT_UI_SERVE_BASE}/ui-settings`

Reply:
BASE_URL is also used for the router history. Any change needed there?

Reply:
Sorry, missed this before - shouldn't be any change needed there @pleek91